lib/blockchain/getTokenSupplyAmount.ts

```
@@ -0,0 +1,29 @@
+import { getChainById } from 'connectors/chains';
+import { ethers } from 'ethers';
+
+import { getTokenDecimals } from './getTokenDecimals';
+
+export async function getTokenSupplyAmount({
+  chainId,
+  tokenContractAddress
+}: {
+  tokenContractAddress: string;
+  chainId: number;
+}) {
+  const tokenDecimal = await getTokenDecimals({ chainId, tokenContractAddress });
+  const chain = getChainById(chainId);
+  if (!chain || tokenDecimal === null) {
+    return null;
```

this should probably throw an error?

---

Yeah sure, added a log to keep track of the chain id and token address combo